routemaster/controllers/parser.rb

```diff
@@ -0,0 +1,38 @@
+require 'routemaster/controllers'
+require 'routemaster/mixins/log'
+require 'sinatra/base'
+
+module Routemaster
+  module Controllers
+    module Parser
+      module Helpers
+        include Mixins::Log
+
+        def data
+          @parsed_data
+        end
+      end
+
+      def self.registered(app)
+        app.helpers Helpers
+
+        app.set :parse do |format|
+          condition do
+            case format
+            when :json
+              begin
+                @parsed_data = JSON.parse(request.body.read)
```

Comment:
I didn't on purpose — nothing else should read the body, otherwise we'd have a wasteful double-read. Good point though, thanks.